include/graphblas/reference/blas3.hpp

```diff
@@ -57,6 +57,25 @@
 		"********************************************************************" \
 		"******************************\n" );
 
+#ifndef _DEBUG_PRINT
+	#ifndef _DEBUG
+		#define _DEBUG_PRINT( msg )
+	#else
+		#ifdef _GRB_WITH_OMP
+			#ifndef _GRB_DEBUG_CRITICAL_SECTION
+				#define _GRB_DEBUG_CRITICAL_SECTION _Pragma("omp critical(_GRB_DEBUG_CRITICAL_SECTION)")
+			#endif
+			#define _DEBUG_PRINT( msg ) \
+				_GRB_DEBUG_CRITICAL_SECTION \
+				{ \
+					std::cout << "[T" << omp_get_thread_num() << "] - " << msg << std::flush; \
+				}
+		#else
+			#define _DEBUG_PRINT( msg ) std::cout << msg << std::flush;
+		#endif
+	#endif
+#endif
```

Comment:
This is a tricky thing to add -- if it's here, maybe it should be used everywhere in reference and beyond

If adding it everywhere, I'm not that keen on inserting many critical sections as part of debugging.

If this is to become a logger mechanism, it clashes with one that Alberto was also working on

If this is instead only meant for this file only, would it not be better to just use the current `#ifdef _DEBUG` everywhere? (Is it that critical(;) to have the thread number and critical sections for the functions in this file?)

Reply:
Will remove, we'll adapt using Alberto's work